pkg/operator/finalizer.go

```diff
@@ -66,7 +65,7 @@ func (c *Controller) finalizeResources(o *imageregistryv1.Config) error {
 	cr := o
 	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 		if cr == nil {
-			cr, err := client.Configs().Get(o.Name, metav1.GetOptions{})
+			cr, err := c.listers.Config.Get(o.Name)
```

Comment:
when we're try to fetch the latest object because of a conflict, we probably want to go to the souce, not the cache.

Reply:
The retry is technically for the create/update that happens towards the end of the function, not the Get, as I don't believe that would return a conflict type error.

Reply:
yes but the point of this get is to ensure we have the latest version of the object before we start making changes and invoke update on it, right?

if the update fails due to a conflict we'll loop back up to here, fetch a "newer" version of the object and try again... except if now if the cache isn't updated, the "newer" version won't be any newer and we'll fail on the update again, until the cache is updated.

Reply:
updated per our conversation